src/Qir/Execution/Tools/QirTools.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Quantum.Qir.Tools.Executable;
+using Microsoft.Quantum.QsCompiler;
+using Microsoft.Quantum.QsCompiler.QIR;
+
+namespace Microsoft.Quantum.Qir.Tools
+{
+    /// <summary>
+    /// Provides high-level utility methods to work with QIR.
+    /// </summary>
+    public static class QirTools
+    {
+        /// <summary>
+        /// Creates a QIR-based executable from a .NET DLL generated by the Q# compiler.
+        /// </summary>
+        /// <param name="qsharpDll">.NET DLL generated by the Q# compiler.</param>
+        /// <param name="libraryDirectories">Directory where the libraries to link to are located.</param>
+        /// <param name="includeDirectories">Directory where the headers needed for compilation are located.</param>
+        /// <param name="executablesDirectory">Directory where the created executables are placed.</param>
+        public static async Task BuildFromQSharpDll(
+            FileInfo qsharpDll,
+            IList<DirectoryInfo> libraryDirectories,
+            IList<DirectoryInfo> includeDirectories,
+            DirectoryInfo executablesDirectory)
+        {
+            using var qirContentStream = new MemoryStream();
+            if (!AssemblyLoader.LoadQirByteCode(qsharpDll, qirContentStream))
+            {
+                throw new ArgumentException("The given DLL does not contain QIR byte code.");
+            }
+            
+            if (!executablesDirectory.Exists)
+            {
+                executablesDirectory.Create();
+            }
+            
+            var tasks = EntryPointOperationLoader.LoadEntryPointOperations(qsharpDll).Select(entryPointOp =>
```

Comment:
Not immediately clear if all the builds will run at the same time.

Reply:
Added a comment about it.